app/lib/identity_tijuana/member_sync.rb

```diff
@@ -78,6 +147,88 @@ def self.import_user(user_id, sync_id)
+    #  3. The first remaining user
+    def self.find_primary_user_for_member(member)
+      external_ids = member.member_external_ids.where(system: 'tijuana')
+      if external_ids.nil? || external_ids.count == 0
+        raise "Id member has no linked TJ users"
+      end
+
+      # Some scenarios to think about before changing this:
+      #
+      # (1) A member email address may not match a TJ user email
+      # address. This could be because of:
+      #    * The cleaner fixed a typo
+      #    * Un-normalised data in both (example@gmail.com vs
+      #      Exam.ple@gmail.com
+      #    * The member's email was changed through Id CRM UI
+      #
+      # (1) A member's email address changed in Id, as a result no
+      # associated TJ users have the same email address.
+      #
+      # (2) A member subscribes and becomes active. They then typo
+      # their email when signing a petition from a new browser. The
+      # newer typo'ed user is not yet active (although it will become
+      # active tomorrow morning) but it is subscribed.
+      #
+      # (3) A member has two users associated with it, one with the
+      # correct email and one with a typo. The typo'ed user gets
+      # updated because they sign another petition with the same typo
+      # and a new postcode.
+      #
+      # (4) A member subscribes using a typo'ed email, becomes active
+      # since they are a new member but unsubscribed because their
+      # email bounced. Their Id member record has the correct email
+      # because the Cleanser fixed it. Thus the member and associated
+      # user have different email addresses. The member later signs
+      # another petition using the correct email address.
+
+      # The points above means that we should matching based on email
+      # first if present, otherwise fall back looking at subscribed
+      # users, then users that first created (assuming most members
+      # will likely not typo their email address).
+      #
+      # In particular we can't rely on most-recently-updated, since
+      # typo'ed users can still get updated because of (3) above. We
+      # also can't rely on active, since it's laggy but also typo'ed
+      # users can be active after they first sign up.
+
+      ext_ids = external_ids.map { |id| id.external_id.to_i }
+      users = User.where(id: ext_ids).order(created_at: :asc)
+      if users.count != external_ids.count
+        raise "Not enough matching users for external ids: #{ext_ids}"
+      end
+
+      matching_emails = users.select { |user|
+        # Need to normalise the email to match the normalisation Id
+        # would do, but as-is, i.e. without any typo correction
+        #
+        # XXX shamefully copied from Cleanser
+        user_email = user.email.downcase
+        user_email = user_email.gsub(/\s/, '')
+        user_email = user_email.gsub('%40', '@') # remove html entities
+
+        user_email == member.email
```

Comment:
Not sure how this would ever happen, ie. as mentioned in the comment
> as-is, i.e. without any typo correction

If the member's email was already cleansed, including fixing the typos, then 
```ruby
user_email == member.email # is always false
```

Reply:
Pretty easy:

If `user.email` is `Alice@gmail.com` and `member.email` is `alice@gmail.com` (as it would be after the email was first cleansed) then that expression will be true, as it should be.

If however `user.email` is `Alice@gamil.com` (note the typo) and `member.email` is `alice@gmail.com` (as it would be after the email was first cleansed) then that expression will be false, as it should be.

Reply:
@ivanb777 does this address the concern?

Reply:
oh yeas! armhph... I was misreading the intent here... ultimately, if the original user email was typo'ed, we will then be matching the `primary` based on the TJ user record that was created with a cleansed email.